apprise/py3compat/asyncio.py

```diff
@@ -25,7 +25,6 @@
 
 import sys
 import asyncio
-from concurrent.futures import ThreadPoolExecutor
```

Comment:
The only catch with this is Home Assistant needed it to suppresses a warning.   I wonder if the `async` flag should change from a `True` and `False` to more of an _ENUM_ type, where `True` and `False` are still accepted (As the old types for backwards compatibility), but then allows us to introduce a new type that exercises what you're doing here. Thoughts... something like:
```python
class AsyncMode(object):
    # Backwards compatibility
    OFF = False
    THREADED = True

    # this can be anything.. None might not be the best choice
    EXECUTOR = None
```

Then setting the `async` flag would accept the 3rd state? The problem goes back to the original issue and that is the underlining async calls aren't really pure async compatible (like `requests`) and some calls to `sleep`.  Stuff like that needs to be made more compatible (hence `asyncio.sleep`, and [aoihttp](https://docs.aiohttp.org/en/stable/).  But this won't solve other binary protocols that don't have an async alternative. This is a BIG task/overhaul but probably needed down the road at some point.

I realize i'm rambling a bit here, but i hope what i'm saying makes sense?

 I do prefer the clean `gather()` routine; i'm just concerned about backwards compatibility. I'd rather introduce a new feature and then slowly allow people to flip... eventually deprecating the old way.  I also realize Python v2.7 support is annoying and makes things a bit more complicated.  I'll drop it soon; i'm just an old guy who loves being able to use this on older CentOS boxes. :wink:

Reply:
I don't have Home Assistant, so I haven't tested, but I would not expect that warning to show with this PR. This change accomplishes the same thing as ThreadPoolExecutor - running the synchronous IO-bound stuff inside of an executor that doesn't block the main event loop - without spinning up a new ThreadPoolExecutor for each notification.

Do let me know if that's not the case!